### PR TITLE
update/cancel fee orders when their corresponding orders are canceled

### PR DESCRIPTION
**Pull request by allenjprice**

### CODE REVIEW

**Comment by allenjprice** [comment]
@jalehman trigger warning: there's some real nasty stuff in that second commit.

File: src/clj/odin/graphql/resolvers/order.clj [hunk 1]
```diff
@@ -352,12 +364,40 @@
         (d/entity (d/db conn) id)))))
 
 
+(defn- update-order-attached-tx
+  [fee-order original-order-id]
+  [:db/retract (td/id fee-order) :order/attached (td/id original-order-id)])
+
+(defn- generate-fee-cancellation-tx
+  [conn original-order-id]
+   (let [fee-orders (:order/_attached (d/entity (d/db conn) original-order-id))]
```
Comment (jalehman) [inline src/clj/odin/graphql/resolvers/order.clj]: indentation

File: src/clj/odin/graphql/resolvers/order.clj [hunk 2]
```diff
@@ -352,12 +364,40 @@
         (d/entity (d/db conn) id)))))
 
 
+(defn- update-order-attached-tx
+  [fee-order original-order-id]
+  [:db/retract (td/id fee-order) :order/attached (td/id original-order-id)])
+
+(defn- generate-fee-cancellation-tx
+  [conn original-order-id]
```
Comment (jalehman) [inline src/clj/odin/graphql/resolvers/order.clj]: pass `db`, not `conn`

**Comment by allenjprice** [comment]
the provided code from our review chat last week worked mostly well, with only one small change.

File: src/clj/odin/graphql/resolvers/order.clj [hunk 3]
```diff
@@ -178,19 +179,31 @@
   (* (/ (:service/price fee) 2) (inc occurences)))
 
 
+(defn- get-attached-orders
+  [db fee-id orders]
+  (->> (filter
+        (fn [order]
+          (let [fees (:service/fees (d/entity db (order/service order)))]
```
[the remark this answers is not in the document]
Comment (allenjprice) [reply src/clj/odin/graphql/resolvers/order.clj]: turns out - we do have a `service/fees` selector.

fun fact: `order/service` does _not_ appear to produce an entity! it certainly stands to reason that it should, though

**Pull request merged**